src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -290,16 +292,33 @@ export default class DefaultAudioVideoController implements AudioVideoController
           ]),
           this.configuration.connectionTimeoutMs
         ),
+        new ValidateAttendeePresenceTask(this.meetingSessionContext),
       ]).run();
       this.sessionStateController.perform(SessionStateControllerAction.FinishConnecting, () => {
         this.actionFinishConnecting();
       });
     } catch (error) {
       this.sessionStateController.perform(SessionStateControllerAction.Fail, async () => {
-        const status = new MeetingSessionStatus(
+        let status = new MeetingSessionStatus(
           this.getMeetingStatusCode(error) || MeetingSessionStatusCode.TaskFailed
         );
         await this.actionDisconnect(status, true);
+
+        // TODO: Remove this workaround once the root cause is identified.
+        // Assume you haven't allowed microphone before and joined the session with an empty audio device.
+        // In Safari, SDK doesn't receive an attendee presence event. In Firefox, no ice candidates were gathered.
+        if (
+          this.meetingSessionContext.browserBehavior.requiresGetUserMediaForNoAttendeePresence() &&
+          (status.statusCode() === MeetingSessionStatusCode.NoAttendeePresent ||
+            (error && error.message && error.message.includes('no ice candidates were gathered')))
+        ) {
+          try {
+            await navigator.mediaDevices.getUserMedia({ audio: true });
```

Comment:
I just tried using the media stream broker. It doesn't work because `acquireAudioInputStream` also creates and returns an empty audio device. The `getUserMedia` doesn't get called ([here](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/devicecontroller/DefaultDeviceController.ts#L522)).